src/components/expanded-state/custom-gas/FeesPanel.js

```diff
@@ -0,0 +1,511 @@
+    const maxFee = get(selectedGasFee, 'gasFee.maxFee.native.value.display', 0);
+
+    const currentBaseFee = get(currentBlockParams, 'baseFeePerGas.gwei', 0);
+    let maxBaseFee;
+    if (selectedOptionIsCustom) {
+      // block more than 2 decimals on gwei value
+      const decimals = Number(customMaxBaseFee) % 1;
+      maxBaseFee =
+        `${decimals}`.length > 4
+          ? parseInt(customMaxBaseFee, 10)
+          : customMaxBaseFee;
+    } else {
+      maxBaseFee = parseInt(
+        get(selectedGasFee, 'gasFeeParams.maxFeePerGas.gwei', 0),
+        10
+      );
+    }
+
+    let maxPriorityFee;
+    if (feesGweiInputFocused) {
+      // block more than 2 decimals on gwei value
+      const decimals = Number(customMaxPriorityFee) % 1;
+      maxPriorityFee =
+        `${decimals}`.length > 4
+          ? Number(parseFloat(customMaxPriorityFee).toFixed(2))
+          : customMaxPriorityFee;
+    } else {
+      maxPriorityFee = get(
+        selectedGasFee,
+        'gasFeeParams.maxPriorityFeePerGas.gwei',
+        0
+      );
+    }
+    return { currentBaseFee, maxBaseFee, maxFee, maxPriorityFee };
+  }, [
+    selectedGasFee,
+    currentBlockParams,
+    selectedOptionIsCustom,
+    feesGweiInputFocused,
+    customMaxBaseFee,
+    customMaxPriorityFee,
+  ]);
+
+  const renderRowLabel = useCallback(
+    (label, type, error, warning) => {
+      const openGasHelper = () => {
+        Keyboard.dismiss();
+        navigate(Routes.EXPLAIN_SHEET, {
+          currentBaseFee,
+          currentGasTrend,
+          type,
+        });
+      };
+      let color;
+      let text;
+      if ((!error && !warning) || !selectedOptionIsCustom) {
+        color = isDarkMode
+          ? colors.alpha(darkModeThemeColors.blueGreyDark, 0.25)
+          : colors.alpha(colors.blueGreyDark, 0.25);
+        text = '􀅵';
+      } else if (error) {
+        color = colors.red;
+        text = '􀁟';
+      } else {
+        color = colors.yellowFavorite;
+        text = '􀇿';
+      }
+
+      return (
+        <PanelColumn>
+          <ButtonPressAnimation onPress={openGasHelper}>
+            <Row>
+              <PanelLabel>
+                {`${label} `}
+                <Label color={color} size="smedium" weight="bold">
+                  {text}
+                </Label>
+              </PanelLabel>
+            </Row>
+          </ButtonPressAnimation>
+        </PanelColumn>
+      );
+    },
+    [
+      currentBaseFee,
+      currentGasTrend,
+      navigate,
+      selectedOptionIsCustom,
+      isDarkMode,
+    ]
+  );
+
+  const formattedBaseFee = useMemo(
+    () => `${parseInt(currentBaseFee, 10)} Gwei`,
+    [currentBaseFee]
+  );
+
+  const handleOnInputFocus = useCallback(() => {
+    if (isEmpty(gasFeeParamsBySpeed[gasUtils.CUSTOM])) {
+      const gasFeeParams = gasFeeParamsBySpeed[selectedGasFeeOption];
+      updateToCustomGasFee({
+        ...gasFeeParams,
+        maxBaseFeePerGas: gasFeeParams.maxFeePerGas,
```

Comment:
line 238: `maxBaseFeePerGas` is never actually used and is not part of the `GasFeeParams` entity - can remove this line

line 239: since you are already doing the deconstruction `...gasFeeParams` on line 237, you do not need this line either

Reply:
@estebanmino